ocs_ci/ocs/resources/bucket_notifications_manager.py

```diff
@@ -0,0 +1,362 @@
+
+        try:
+            patch_params = [
+                {
+                    "op": "replace",
+                    "path": NOTIFS_YAML_PATH_NB_CR,
+                    "value": None,
+                },
+            ]
+            self.nb_config_resource.patch(
+                params=json.dumps(patch_params),
+                format_type="json",
+            )
+        except CommandFailed as e:
+            if "not found" in str(e).lower():
+                logger.info("The bucketNotifications field was not found")
+            else:
+                logger.error(f"Failed to disable bucket notifications: {e}")
+                raise e
+        wait_for_pods_to_be_running(
+            pod_names=[constants.NOOBAA_CORE_POD],
+            timeout=60,
+            sleep=10,
+        )
+        logger.info("Bucket notifications have been disabled")
+
+    def create_kafka_topic(self, topic_name=""):
+        """
+        Create a Kafka topic
+
+        Args:
+            name(str|optional): Name of the Kafka topic
+
+        Returns:
+            str: Name of the created Kafka topic
+        """
+        topic_name = topic_name or create_unique_resource_name(
+            resource_description="nb-notif", resource_type="kafka-topic"
+        )
+        topic = self.amq.create_kafka_topic(topic_name)
+        self.kafka_topics.append(topic)
+        return topic_name
+
+    def create_kafka_conn_secret(self, topic):
+        """
+        Create secret from a JSON file that
+        defines the Kafka connection configuration
+
+        Args:
+            topic(str): Name of the Kafka topic
+
+        Returns:
+            secret_ocp_obj: OCP instance of the created secret
+            conn_file_name: Name of the JSON file
+        """
+        conn_name = create_unique_resource_name(
+            resource_description="nb-notif", resource_type="kafka-conn"
+        )
+        secret_name = conn_name + "-secret"
+        conn_file_name = ""
+
+        kafka_conn_config = {
+            "metadata.broker.list": "my-cluster-kafka-bootstrap.myproject.svc.cluster.local:9092",
+            "notification_protocol": "kafka",
+            "topic": topic,
+            "name": conn_name,
+        }
+
+        with tempfile.NamedTemporaryFile(
+            mode="w+", prefix="kafka_conn_", suffix=".json", delete=True
+        ) as conn_file:
+            conn_file_name = os.path.basename(conn_file.name)
+            conn_file.write(json.dumps(kafka_conn_config))
+            conn_file.flush()  # Ensure that the data is written
+
+            OCP().exec_oc_cmd(
+                f"create secret generic {secret_name} --from-file={conn_file.name} -n {self.namespace}"
+            )
+
+        secret_ocp_obj = OCP(
+            kind="secret",
+            namespace=self.namespace,
+            resource_name=secret_name,
+        )
+        self.conn_secrets.append(secret_ocp_obj)
+        return secret_ocp_obj, conn_file_name
```

Comment:
If we are keeping file for further use, Please set `delete=False` in `NamedTemporaryFile` on line 216

Reply:
We don't - we just need its name. 

After creating a secret from the file and setting it in the NooBaa CR, its pods mount the secret under the original file's name. We then need to refer to this name when applying `put-bucket-notification` on the bucket.

It's weird setup, and I've already expressed some concerns on the matter: https://issues.redhat.com/browse/DFBUGS-947